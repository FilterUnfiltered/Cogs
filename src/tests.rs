#[test]
fn test_cogs() {
    let _ = crate::init_tracing();
    let tests = 1..=1;
    for test_index in tests {
        let name = format!("tests/{}.cog", test_index);
        let file = std::fs::read_to_string(&name).unwrap();
        let ast = crate::parse_cog(file, &name).unwrap();
        insta::with_settings!({ snapshot_suffix => format!("{test_index}") }, {
            insta::assert_debug_snapshot!(ast);
            insta::assert_snapshot!("codegen", cogs_codegen::generate(&ast).unwrap());
        });
    }
}
