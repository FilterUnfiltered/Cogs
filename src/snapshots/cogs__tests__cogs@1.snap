---
source: src/tests.rs
expression: ast
snapshot_kind: text
---
Component {
    elements: [
        Html(
            HtmlTag {
                tag: "body",
                attributes: [],
                content: [
                    Html(
                        HtmlTag {
                            tag: "h1",
                            attributes: [],
                            content: [
                                Text(
                                    "Yo.",
                                ),
                            ],
                        },
                    ),
                    Html(
                        HtmlTag {
                            tag: "a",
                            attributes: [
                                Attribute {
                                    name: Text(
                                        "src",
                                    ),
                                    value: Some(
                                        Text(
                                            "https://www.youtube.com/watch?v=dQw4w9WgXcQ",
                                        ),
                                    ),
                                },
                            ],
                            content: [
                                Text(
                                    "Click this",
                                ),
                            ],
                        },
                    ),
                    Block(
                        CodeBlock {
                            content: [
                                Text(
                                    "println!(\"test\");",
                                ),
                                Html(
                                    HtmlTag {
                                        tag: "p",
                                        attributes: [],
                                        content: [
                                            Text(
                                                "More Html",
                                            ),
                                        ],
                                    },
                                ),
                                Text(
                                    "\n    ",
                                ),
                            ],
                        },
                    ),
                    Text(
                        "\n",
                    ),
                ],
            },
        ),
        Text(
            "\n",
        ),
    ],
}
